crates/core/src/sql/compiler.rs

```diff
@@ -369,6 +370,41 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn compile_eq_identity_address() -> ResultTest<()> {
+        let (db, _tmp) = make_test_db()?;
+
+        // Create table [test] without any indexes
+        let schema = &[("identity", Identity::get_type()), ("address", Address::get_type())];
+        let indexes = &[];
+        db.create_table_for_test("test", schema, indexes)?;
+
+        let tx = db.begin_tx();
+        // Compile query, check for both hex formats...
+        let sql = &format!(
+            "select * from test where identity = 0x{} AND identity = {} AND address = {} AND address = 0x{}",
+            Identity::__dummy(),
+            Identity::__dummy(),
+            Address::__DUMMY,
+            Address::__DUMMY,
+        );
+        let CrudExpr::Query(QueryExpr {
+            source: _,
+            query: mut ops,
+        }) = compile_sql(&db, &tx, sql)?.remove(0)
+        else {
+            panic!("Expected QueryExpr");
+        };
+
+        assert_eq!(1, ops.len());
+
+        // Assert no index scan
+        let Query::Select(_) = ops.remove(0) else {
+            panic!("Expected Select");
+        };
+        Ok(())
+    }
```

Comment:
can we have some tests that successfully get rows, too, to make sure it works?

Reply:
Done